fix: added missing back buttons to settings pages

al-rosenthal opened the pull request:
# Summary of Changes

updated header for biometry, update pin, terms and push notifications screens in the settings stack

# Screenshots, videos, or gifs

n/a

# Breaking change guide

n/a

# Related Issues

[BC Wallet: 2338](https://github.com/bcgov/bc-wallet-mobile/issues/2338)

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this)
- [x] If applicable, screenshots, gifs, or video are included for UI changes
- [x] If applicable, breaking changes are described above along with how to address them
- [x] Updated documentation as needed for changed code and new or modified features
- [x] Added sufficient [tests](../packages/legacy/core/__tests__/) so that overall code coverage is not reduced

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated checks have passed

al-rosenthal committed:
fix: added missing back buttons to settings pages

updated header for biometry, update pin, terms and push notifications screens in the settings stack

Signed-off-by: al-rosenthal <alfred.k.rosenthal@gmail.com>

bryce-mcmath reviewed (commented):
Is this event / event emitter used anywhere else in the code base? Is there going to be any dead code because of this change that could be removed?

al-rosenthal committed:
fix: cleaning up old code

Signed-off-by: al-rosenthal <alfred.k.rosenthal@gmail.com>

al-rosenthal committed:
Merge branch 'main' into fix/MissingBackButton

al-rosenthal commented on packages/legacy/core/App/navigators/SettingStack.tsx:
```diff
@@ -109,9 +96,9 @@ const SettingStack: React.FC = () => {
-          headerLeft: biometryUpdatePending ? () => null : undefined,
           headerBackTestID: testIdWithKey('Back'),
           ...ScreenOptionsDictionary[Screens.UseBiometry],
+          headerLeft: undefined,
```
these are overriding the default screen options config so that the back button appears in the navigation

bryce-mcmath replied:
Should we maybe apply these directly to the default screen options dictionary, so that others can change it for themselves if they desire?

al-rosenthal replied:
updated the default config to have the back navigation by default. I've also updated the onboarding stack to remove them as the everyday need is the back button should be present

al-rosenthal commented on packages/legacy/core/App/navigators/SettingStack.tsx:
```diff
@@ -38,16 +35,6 @@ const SettingStack: React.FC = () => {
   const OnboardingTheme = theme.OnboardingTheme
   const carousel = createCarouselStyle(OnboardingTheme)
 
-  useEffect(() => {
```
in this context we don't need a complex control for the navigation button on the biometry page, from the setting screen it should just be present. This does not affect the biometry page during onboarding

bryce-mcmath reviewed (changes requested):
One more question / nitpick, but nice job finding and snipping out that dead code

jleach commented on packages/legacy/core/App/screens/UseBiometry.tsx:
```diff
@@ -2,13 +2,12 @@ import { CommonActions, useNavigation } from '@react-navigation/native'
 import React, { useState, useEffect, useCallback, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
-import { StyleSheet, Text, View, Modal, ScrollView, DeviceEventEmitter, Linking, Platform } from 'react-native'
+import { StyleSheet, Text, View, Modal, ScrollView, Linking, Platform } from 'react-native'
```
Why remove `DeviceEventEmitter`?

al-rosenthal replied:
it wasn't be used anymore. From what I can tell in the code the emitter was being used to control if the navigation button should be present but since the screen props were refactored that isn't needed anymore.

jleach replied:
Okay.

al-rosenthal committed:
refactor: moved screen option responsibility

Moved back navigation special case logic to onboarding flow and left back navigation apart of the default screen options

Signed-off-by: al-rosenthal <alfred.k.rosenthal@gmail.com>

al-rosenthal committed:
Merge branch 'main' into fix/MissingBackButton

bryce-mcmath commented on packages/legacy/app/container-imp.tsx:
```diff
@@ -51,7 +51,7 @@ export class AppContainer implements Container {
   }
 
   public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
-    return this._container.resolve(token) as TokenMapping[K]
```
Why this change? It seems unrelated

al-rosenthal replied:
I've got a sonar cube plugin and flagged the casting as unnecessary

bryce-mcmath commented on packages/legacy/app/container-imp.tsx:
```diff
@@ -23,7 +23,7 @@ export class AppContainer implements Container {
     //This is an example of how to customize the screen layout and use custom header for wallets who wnat to hide default navigation header
     //To hide navigation header for a specific page, use headerShown: false in the screen options like this
-    /*
+    /**
```
Why this change? Left over from when you were thinking of moving this commented example?

al-rosenthal replied:
I decided to leave the comment where it is as it doesn't make much sense in the default config either. My sonar cube plugin also flags this to be deleted but with `/**` it treats it as more than just a code block

bryce-mcmath commented on packages/legacy/core/App/container-impl.ts:
```diff
@@ -207,7 +207,7 @@ export class MainContainer implements Container {
   }
 
   public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
-    return this._container.resolve(token) as TokenMapping[K]
```
Same as above

al-rosenthal replied:
I've got a sonar cube plugin and flagged the casting as unnecessary

bryce-mcmath reviewed (commented):
Just a couple questions, functional code changes look great

al-rosenthal committed:
Merge branch 'main' into fix/MissingBackButton
